src/block/block_compact.c

```diff
@@ -115,7 +121,7 @@ __wt_block_compact_skip(WT_SESSION_IMPL *session, WT_BLOCK *block, int *skipp)
 	 * empty file can be processed quickly, so more aggressive compaction is
 	 * less useful.
 	 */
-	if (avail_ninety >= fh->size / 10) {
+	if (avail_total > WT_MEGABYTE && avail_ninety >= fh->size / 10) {
```

Comment:
Should have read further before replying!

But why check`avail_total`?

If all the space is at the end of the file, it's not going to help to compact, it has to be at the start of the file. How about:

```
/*
 * Skip files where we can't recover at least 1MB.
 *
 * If at least 20% of the total file is available and in the first 80%
 * of the file, we'll try compaction on the last 20% of the file; else,
 * if at least 10% of the total file is available and in the first 90%
 * of the file, we'll try compaction on the last 10% of the file.
 *
 * We could push this further, but there's diminishing returns, a mostly
 * empty file can be processed quickly, so more aggressive compaction is
 * less useful.
 */
if (avail_eighty > WT_MEGABYTE && avail_eighty >= ((fh->size / 10) * 2)) {
        *skipp = 0;
        block->compact_pct_tenths = 2;
} else if (avail_ninety > WT_MEGABYTE && avail_ninety >= fh->size / 10) {
        *skipp = 0;
        block->compact_pct_tenths = 1;
}
```